cli/src/main.rs

```diff
@@ -119,24 +119,27 @@ async fn main() -> Result<(), color_eyre::Report> {
         "Hyperledgerいろは2にようこそ！(translation) Welcome to Hyperledger Iroha!"
     );
 
-    let genesis = if let Some(genesis_path) = &args.genesis_path {
-        GenesisNetwork::from_configuration(
-            args.submit_genesis,
-            RawGenesisBlock::from_path(
-                genesis_path
-                    .first_existing_path()
-                    .ok_or({
-                        color_eyre::eyre::eyre!("Genesis block file {genesis_path:?} doesn't exist")
-                    })?
-                    .as_ref(),
-            )?,
-            Some(&config.genesis),
-            &config.wsv.transaction_limits,
-        )
-        .wrap_err("Failed to initialize genesis.")?
-    } else {
-        None
-    };
+    let genesis =
+        if let Some(genesis_path) = args.submit_genesis.then_some(()).and(args.genesis_path) {
```

Comment:
Marin prefers the imperative style.

Reply:
> Marin prefers the imperative style. 

Actually, i don't care. I prefer that we have a uniform style. But sometimes I dont find jt worth my time to enforce it religiously. Ww need to have a faster turnover of PRs. I think every one of us should ask themselves is this comment really worth delaying the merge? I will always apply every comment if I'm neutral or in favor of it. In this case there was a `?` inside the closure which I didn't want to figure out how. My mistake was that I closed the discussion instead of asking for help

Reply:
I change it to functional style. It was very easy, I was lazy to do it. And it's indeed looking a lot more readable